### ISSUE TITLE
React-Native Accessibility Focus Order Issues 

### ISSUE THREAD

**Issue by Naresh6885** [facebook/react-native#29910]
Hi, I am implemented the accessibility in one of react native application. I am facing the issue related to setting the focus order of screen components. I had done a lot of R & D on this but I didn't found any solution for this.
I am facing the following problems:
1) The focus order is not chronological.
2) when ever the date & time picker appears in app, it is not focusable.
3) If any alert / modal appears then after closing of that alert / modal the focus should one the same component from which the popup was initiated.

Although according to my R & D, I have found:
1) for focusing the components in chronological order, the view hierarchy should be render in the same order, but due to existing & legacy project that's not possible for now.
2) I have found one solution by setting the reactTag of elements & then calling the **AccessibilityInfo.setAccessibilityFocus(reactTag)** that's again creating a lots of issues, the focus gone to other screen elements randomly.

I have reported this on [stack-overflow](well: https://stackoverflow.com/questions/63534817/how-can-i-alter-or-maintain-the-focus-order-of-views-in-react-native-accessibil)

React Native Environment Info:
    System:
      OS: macOS 10.15.3
      CPU: (4) x64 Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
      Memory: 28.02 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 8.13.0 - /usr/local/bin/node
      Yarn: 1.12.3 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
      Android SDK:
        API Levels: 23, 25, 26, 27, 28, 29
        Build Tools: 26.0.2, 26.0.3, 27.0.1, 27.0.3, 28.0.2, 28.0.3, 29.0.2
        System Images: android-26 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.6 => 16.8.6 
      react-native: 0.59.5 => 0.59.5 
    npmGlobalPackages:
      react-native-cli: 2.0.1

**Comment by ZioVio** [comment]
Hi there, I am facing the same problem. According to [this](https://github.com/facebook/react-native/issues/30097) issue we tried to use ```AccessibilityInfo.setAccessibilityFocus``` several times but it leads to the mess where we have to call for focus for 4+ times. I don't think it's a good solution. I wish that guys from FB paid their attention to this problem

**Comment by unstoo** [comment]
Hi everyone, was this issue addressed? I'm working on a big app that provides medical services. This question is causes a lot of pain.

**Comment by ZioVio** [comment]
I can share my `react-native info` as well if it helps somehow. We are still facing this issue and will be very happy if the solution appears

```
System:
    OS: macOS 11.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 88.02 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.4.2 - /usr/local/bin/node
    Yarn: Not Found
    npm: 8.1.3 - /usr/local/bin/npm
    Watchman: 2021.06.07.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK:
      API Levels: 23, 28, 29, 30
      Build Tools: 28.0.3, 29.0.0, 29.0.2, 30.0.2, 30.0.3
      System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom_64, android-30 | Google Play Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.2 AI-202.7660.26.42.7486908
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.0 => 0.66.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by blavalla** [comment]
Does anyone have a code example of this problem? It could be caused by a number of different factors, and it'll be hard to narrow down without a reproducible case. Thanks!
